### DISCUSSION TITLE
Discord community call: 12 October 2024

### DISCUSSION THREAD

**Discussion by BenJamesBen** [freesewing/freesewing#7117] [Support] [answered]
You are all invited to our Discord hangout.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This page will be updated after the call with notes.

### About

FreeSewing's Discord hangout is an informal hangout/chat/meeting that is open to everyone.
The call is organized/hosted by @tangerineshark.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

### Details

Location: https://discord.freesewing.org/ (meeting room: 🔊voice chat)
Date: October 12th, 2024 4pm UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/oct-12-2024/4pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.

### Agenda

- See comments below

**Comment by BenJamesBen** [comment]
# FreeSewing documentation transition to Docusaurus.

What happened?
- We switched to using Docusaurus to generate our freesewing.dev website.
- The transition already occurred on 2024-09-24 for freesewing.dev
- Some dev documentation was removed, too.
- Some dev documentation was altered slightly, too.

What does Docusaurus do? 
- It generates and runs the website that displays documentation.
- Handles headers, footers, top navigation menu, sidebar navigation menu, etc.
- Handles formatting for page components, etc.

Why did we switch to Docusaurus?
- Previously Joost did all the work of writing and maintaining the previously-mentioned stuff.
- This was work just to run the freesewing.dev website rather than on the actual FreeSewing project.

What do doc writers need to know?
- Much of the documentation files are unchanged. The files are still written in Markdown. Many were just ported over as-is or with minor tweaks.
- Dev documentation files were moved to a new location in the repository:
  - From: `markdown/dev`
  - To: `sites/dev/docs`
- New filename for directory index pages:
  - Old: `en.md` 
  - New: `readme.mdx`
- New FrontMatter is supported by Docusaurus and used by us in a few places:
  - `sidebar_position: 40`
  - `sidebar_label: Frequently Asked Questions`
- New Docusaurus "Admonitions" replace old FreeSewing custom tag blocks:
  - Old:: `<Related>`, `</Related>`
  - New: `:::note RELATED`, `:::`
  - Types of Admonitions: `note`, `tip`, `info`, `warning`, `danger`
  - One-line admonitions aren't allowed. Make sure the closing `:::` is on a separate line.
- Two FreeSewing custom tag blocks still remain: `<Example>`, `<ReadMore>`

The directory structure was flattened in many cases.
- Old: multiple subdirectories, each subtopic in its own subdirectory, each with its own little `en.md` file. Many little subdirectories with many little files.
- New: one single `readme.mdx` file with all the info in the one file.
- For example, Design Best Practices Guide.
- Reduces complexity.

What documentation was removed?
- Top-level doc that mainly just provided links to other sections in the docs: "Ways to Contribute", "Design" overview, "Support FreeSewing"
- the sitemap
- i18n documentation
- Guides: Unused-never-written content guides (MDX, Sanity), "How We Structure our Documentation", Translation Guide, Markdown Guide
- Howtos: "Ways to Contribute"
- Reference: Sites (Sanity info), Terms (CJS, ESM, Variadic), Web of Trust, Backend API (but the Swagger docs still exist)

Questions I do not yet have answers for.
- ~Will freesewing.org also switch over to Docusaurus? If so, when? And, what parts of the site will switch over?~
- ~Will a search feature be added to freesewing.dev? (You can no longer search the site.)~
- ~Will we be re-adding documentation for the Backend API? Or, will we instead just use and maintain the Swagger doc?~

---

## For people who want more technical details.

You can still run a development environment version of freesewing.dev
- `cd sites/dev ; yarn start`
- The Docusaurus site runs on port 3000, not 8000.
- http://localhost:3000

Docusaurus references
- https://docusaurus.io/
- docs: https://docusaurus.io/docs
- admonitions: https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs

New config files:
- `sites/dev/docusaurus.config.mjs`
- `sites/dev/sidebars.js`
- `sites/dev/babel.config.js`

Other directories of interest:
- /dev/src contains code for FreeSewing's custom tags, CSS theming, the freesewing.dev home page
- /dev/static contains images, logos, icons

New technologies used under the covers:
- Prism: Styles/highlights code syntax. https://prismjs.com/
- Babel: Compiles JavaScript code examples into simpler JS that browsers can run. https://babeljs.io/

**Reply by joostdecock** [reply]
You're the best Ben 🙏

Let me try to answer your remaining questions:


**Will freesewing.org also switch over to Docusaurus? If so, when? And, what parts of the site will switch over?**

My plan (I'm already working on it) is to move the documentation to Docusaurus, and then anchor that site under the `/docs` path of FreeSewing.org.

In other words, FreeSewing.org would become a _frankensite_ composed of different purpose-built minisites that are glued together in a way that is transparant to the user.

The `showcase` and `blog` content can also be handled by Docusaurus and would be mapped to `/showcase` and `/blog` respectively. 

The account management and pattern generation would then be the remaining part, which would essentially be a new `lab` (a site you can run in the monorepo to test patterns.

By pulling the site apart into different sections like this, we should be able to focus in the things that make FreeSewing FreeSewing, and I also think it will make it easier for contributors to get involved with this aspect of the project.

**Will a search feature be added to freesewing.dev? (You can no longer search the site.)**

Yes, but rather than maintain it ourselves, I have applied for [Algilolia's docsearch initiative](https://docsearch.algolia.com).

Once we've been approved (should not be a problem) I will add search to the site.
To apply, the site needs to be up, so I had no choice but to deploy it without search before I could make the switch.

***Will we be re-adding documentation for the Backend API? Or, will we instead just use and maintain the Swagger doc?**

Both. However, I feel the backend needs a bit of work, and the documentation has slowly gotten out of sync. So I plan to revisit it and update the documentation after completing my current work on the websites.

I've removed the docs in the meanwhile because they used a bunch of custom components that I would have to port to Docusaurus to as the docs as they are.
So, I decided I'd rewrite they later into something easier to maintain.


Hope this helps. Yhank you for your stellar work on clarifying all this. Please don't hesitate to reach out if you have questions.

**Reply by joostdecock** [reply]
Perhaps one remark about the admonitions.
As you mentioned, Docusaurus supports the following admonitions: `note`, `tip`, `info`, `warning`, `danger`.

A lot of thise match exactly to the custom components we used before, eg a tip:

```
:::tip
Your tip here
:::
```

However, you can also override the title of an admonition, that is how we use them for something like `related`:

```
:::tip RELATED
Something related 
:::
```

This will be styled as `tip` but the title will be `RELATED` (Note that the title will always be uppercased even if not written in uppercase)

**Comment by joostdecock** [comment]
Quick update with more Docusaurus-related news:

**What' we're doing**

Copy/paste from Discord:

> Essentially, the idea is that we'll have a few mini-sites:
>
> - dev docs (Docusaurus): published at FreeSewing.dev
> - org docs (Docusaurus): published at freesewing.org/docs
> - org blog (Docusaurus): published at FreeSewing.org/blog
> - org showcase (Docusaurus): published at FreeSewing.org/showcase 
> - app (NextJs): published at FreeSewing.org
> 
> So we will focus on account management and pattern generation and leave handling everything that is markdown/MDX to Docusaurus. That will simplify things a lot, and we can build/deploy each part individually and glue them together using Vercel's multi-zone feature: https://nextjs.org/docs/pages/building-your-application/deploying/multi-zones

**FreeSewing.dev**

I've (re-)added the search functionality to FreeSewing.dev, as that was missing after the migration.

**FreeSewing.org**

I have stared the process of migrating the documentation of FreeSewing.org to Docusaurus. 
It's a work in progress, but available at: https://docs.freesewing.org/ 

What's still to be done is the designs documentation.
I have ported both `aaron` and `brian`, but the other designs are still to be done.

I could really use some help with this, so here's what needs to happen for each design:

- Docs go in `sites/orgdocs/docs/designs/` (check the `aaron` and `brian` examples there.
- Rename the `en.md` to `readme.mdx`, and get rid of all other languages
- Merge all but _sewing instructions_ and _design options_ into a single page, see the `aaron` example.
- Make sure to port the content and use the admonitions instead of custom components (like `Tip` and so on)
- For the __design options_ page, make sure to apply the same frontmatter as in the `aaron` example, this will cause the design options pages to drop from the sidebar
- Then, compile an umbrella page for all design options (here too, check the example). Make sure to set the header ids to the lowercased name of the option (so we can link to them exact header when we know the option name. For example, if the option is `frontArmholeDeeper` and the title is `Front armhole extra deptch`, make sure to add `{#frontarmholedeeper}` behind the title, as this will set the title ID.

The purpose here is to have all options on 1 page, but still have the per-option pages exist so we can (perhaps) load them in and iframe other places.

Once again, this should be a lot clearer when you check the examples for `aaron` and `brian`.
To run this locally, you enter the `sites/orgdocs` folder and run `npm run start`.

**Comment by BenJamesBen** [comment, marked as answer]
### October 12, 2024 Community Call

#### Notes:
- Overview of switch to Docusaurus.
  - No major changes for people who write/edit documentation. Most of the changes are in the backend.
  - Doc files are in a different place in the repo.
  - The file still use Markdown. 95-98% the same as before.
  - Slightly different syntax for Tips, Notes, Related, etc. ("Admonitions")
  - Flattened doc structure. Instead of a lot of little files and subdirectories, now there are bigger, single files.
  - See: https://github.com/freesewing/freesewing/discussions/7117#discussioncomment-10826030 for details about the Docusaurus changes.
- More information about the switch to Docusaurus.
  - We previously used MDX for documentation and did everything ourselves to maintain the website. Now Docusaurus exists to do this for us.
  - There are different sub-sites in the backend that look like one single website on the front end.
  - Flattening: The reason why we used to have a lot of little files and subdirectories was a technical limitation of SSR (Server-side Rendering) and how linking worked. But, we don't have the limitation now.
  - However, design options docs have not been flattened, to allow the UI to link to individual doc pages.
- Joost needs help porting documentation for the invidual designs.
  - See Aaron and Brian as examples of what to do. And see https://github.com/freesewing/freesewing/discussions/7117#discussioncomment-10923381 for an overview of the process
  - Are designers expected to go and update the docs for their own designs? No. But, all help is welcome.
- Curvalicia needs help pattern testing Eve, the sleeve.
- Translations: The translation for useful UI terms have been compiled into Markdown tables, in case we want to use them for the official docs or unofficial wiki.
- How to make FreeSewing patterns easier for laser-cutter users. 
  - Currently, there is a lot of post-processing that needs to happen to patterns before they are ready for laser cutting.
  - Outline on seam allowance lines? Yes, but perforations can also be useful on the sewing lines.
  - Cannot/difficult to cut-on-fold using a laser cutter. A large part of the pattern-processing effort is to unfold/mirror cut-on-fold parts.
  - Notches. Traditional notches are triangles in the seam allowance, and this works well for laser cutters. However, FreeSewing's notches are on the seam line.
- Why does Brian still use Biceps Ease in addition to Waist-to-Armpit Depth and Armhole Depth for the v3 armscye calculations?
  - Wasn't the v3 change supposed to remove both Biceps Circumference and Biceps Ease as factors?
  - Biceps Ease makes sleeve bigger, and so the armscye also needs to get bigger.
  - More pattern testing could help.
- Curated measurements sets. We need more.
  - Getting new measurements sets from YouTube sewing channel celebs?
  - What is the process for submitting/suggesting new measurements sets? A submission made 6 months ago has been ignored.
  - A lot of the process is now done by Joost. He'll try to document it so others can do this work. Guidelines for what photos to submit.

#### Attendees:
- Benjamin F.
- bobgeorgethe3rd
- Curvalicia (a.k.a. Anna Puk)
- joostdecock
- Leviathan
- RockerKitten
- Thrunic
- Wouter.vdub
